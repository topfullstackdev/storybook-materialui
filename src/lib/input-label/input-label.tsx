import {
  InputLabel as MUIInputLabel,
  InputLabelProps as MUIInputLabelProps,
  css,
} from '@mui/material';

export interface InputLabelProps extends MUIInputLabelProps {
  fontColor?: string;
}

export function InputLabel({ children, fontColor, ...props }: InputLabelProps) {
  return (
    <MUIInputLabel
      css={css`
        font-size: 1.2rem;
        color: ${fontColor || 'black'};
        z-index: 1;
        width: auto;
        margin-bottom: 5px;
        font-family: Verdana, Arial, sans-serif;
      `}
      {...props}
    >
      {children}
    </MUIInputLabel>
  );
}

export default InputLabel;
