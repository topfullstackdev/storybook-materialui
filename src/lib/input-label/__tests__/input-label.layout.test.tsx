import { render } from '@testing-library/react';

import { InputLabel } from '../input-label';

describe('InputLabel', () => {
  it('should render successfully', () => {
    const { container } = render(<InputLabel>Label Text</InputLabel>);
    expect(container).toMatchSnapshot();
  });
});
