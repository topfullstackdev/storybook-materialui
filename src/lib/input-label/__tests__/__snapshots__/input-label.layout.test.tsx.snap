// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel should render successfully 1`] = `
.emotion-0 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-size: 1.2rem;
  color: black;
  z-index: 1;
  width: auto;
  margin-bottom: 5px;
  font-family: Verdana,Arial,sans-serif;
}

.emotion-0.Mui-focused {
  color: #1976d2;
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-0.Mui-error {
  color: #d32f2f;
}

<div>
  <label
    class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary emotion-0"
  >
    Label Text
  </label>
</div>
`;
